src/main/java/com/google/sps/servlets/NewMatchQueryServlet.java

```diff
@@ -0,0 +1,39 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.algorithms.FindMatchQuery;
+import com.google.sps.data.MatchRequest;
+import com.google.sps.data.User;
+import java.io.IOException;
+import java.util.Collection;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet responsible for displaying new mentor matches after user fills out match form. */
+@WebServlet("/new-matches-query")
+public class NewMatchQueryServlet extends HttpServlet {
+
+  @Override
+  public void init() {}
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Gson gson = new Gson();
+
+    // Convert the JSON to an instance of MatchRequest.
+    MatchRequest matchRequest = gson.fromJson(request.getReader(), MatchRequest.class);
```

Comment:
fromJson could return a null value or throw an exception if the request is not a valid representation of the MatchRequest class. 

Since this servlet will respond to any request (including requests that do not originate from your frontend code), it may be helpful to do some validation on the request before serializing it to a Java object.

Reply:
Added a method to validate the request